api/config/v1alpha1/envoygateway_types.go

```diff
@@ -105,6 +110,14 @@ type EnvoyGatewayFileProvider struct {
 	// TODO: Add config as use cases are better understood.
 }
 
+// XDSServer defines configuration for the xds server.
+type XDSServer struct {
+	// TLSConfig configures the file paths to the tls cert, private key, and ca cert.
+	//
+	// +optional
+	TLSConfig *TLSConfig `json:"tlsConfig,omitempty"`
```

Comment:
recommend looking at existing APIs and reusing them
* https://github.com/envoyproxy/gateway/blob/ea2a688f61451b5cfebf2e23fd6a85efc9d4ea69/api/config/v1alpha1/envoygateway_types.go#L197
* https://github.com/envoyproxy/gateway/pull/1294

The only addition that will be needed here is a new field called `CACertificateRef` which stores the CA Cert as a secret as `tls.crt` in the TLS Type Secret

Reply:
Thanks, I will take a look.

Reply:
While I think it does make sense to use the ExtensionTLS (and underlying [SecretObjectReference](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.SecretObjectReference)) for configuring the envoy proxy tls configuration I am not sure it make sense for the envoy gateway tls certs. The envoy gateway certs are mounted as a volume by helm or manually by the user so I am not sure they need to be a reference to a secret but rather the paths they are expected to be mounted at. What do you think?

Reply:
I see what you mean https://github.com/envoyproxy/gateway/blob/d801f1c7c5152ac68e2c013e9d7f3eab13731d11/charts/gateway-helm/templates/envoy-gateway-deployment.yaml#L93
I prefer if we could find creative ways to leverage using and mounting this new secret that the user provided and under the hood we did the right thing (new crt can be accessed via `/certs/tls.crt`)

Reply:
Just to clarify, you are thinking rather than use the volume mount, a user would configure a `SecretObjectReference` that envoy gateway would then watch/get the latest version via the kube api?

Reply:
close ish, still thinking :)
So I like the idea of a user providing certs/creds as a Secret for K8s, but I also like the idea of the internal process using the creds from the file system, which allows EG to be used in non K8s envs easily.
So should we just add a knob in helm charts that disabled the cert job and instead add docs on how users can add creds as a Secret with a specific name (`envoy-gateway` and `envoy`) in the controller namespace ?
That is what [Istio ](https://istio.io/latest/docs/tasks/security/cert-management/plugin-ca-cert/) does today.

Reply:
Got it, that is pretty close to what I was going for originally. So I think I will move the Envoy Proxy `TLSConfig` over to a `TLSExtention`/`SecretObjectReference`. This will let someone using kubernetes configure a secret for envoy gateway to mount to the Envoy Proxy deployment pods for control plane auth.  Then I will keep something similar to the `TLSConfig` for configuring Envoy Gateway's tls certs. This will allow a user to specify the paths to the tls certs without envoy gateway needing to know how they got there.

I don't think the secret name is import, the user can supply the name of the secret to helm. Although we can default to the current (envoy-gateway and envoy) for backwards compatibility or when the user wants to use the certgen job.

Reply:
Can we start off w/o an APi change and just docs, telling the user to create a specific secret with a specific name, and disable cert job, so when envoy gateway starts, the secret exists  and is mounted in ?

Reply:
Hoping @envoyproxy/gateway-maintainers can share their thoughts as well

Reply:
Even simpler I like it. Yep I can do that 😄

Reply:
Hey @dboslee we discussed this in the community call today with @AliceProxy @haq204 @kflynn and we are in agreement reg 
1. disabling the cert gen job using a helm chart knob
2. documenting how a user can bring their own certs by 
  * creating secrets with specific names in the EG namespace
  * installing EG with the cert gen job disabled

Reply:
Great I will put something together.

Reply:
awesome, thanks

Reply:
@dboslee raised https://github.com/envoyproxy/gateway/issues/1471 to highlight that cert-gen should not overwrite certs if they are already present in the namespace, so the helm knob will likely not be needed

Reply:
Got it so sounds like we just need docs on how to use cert manager?

Reply:
yeah thats right